### PR TITLE
Resolves 398 -> errors set status to known, status details set to error message

### PR DESCRIPTION
**Pull request by JazminElkan-Gonzalez**
Tested by: 

forced  runtime.DefaultUnstructuredConverter.FromUnstructured to return an error

result: health was set to unknown.

### CODE REVIEW

File: controller/appcontroller.go [hunk 1]
```diff
@@ -567,17 +568,18 @@ func setApplicationHealth(comparisonResult *appv1.ComparisonResult) (*appv1.Heal
...
-			}
+			poErr = err
 			resource.Health = *healthState
 		}
 		comparisonResult.Resources[i] = resource
 		if health.IsWorse(appHealth.Status, resource.Health.Status) {
 			appHealth.Status = resource.Health.Status
+			if poErr != nil {
+				savedErr = poErr
+			}
```
Comment (jessesuen) [inline controller/appcontroller.go]: Why do we only set savedErr in the health.IsWorth block?

File: controller/appcontroller.go [hunk 2]
```diff
@@ -567,17 +568,18 @@ func setApplicationHealth(comparisonResult *appv1.ComparisonResult) (*appv1.Heal
 				return nil, err
 			}
 			healthState, err := health.GetAppHealth(&obj)
-			if err != nil {
-				return nil, err
-			}
+			poErr = err
```
Comment (jessesuen) [inline controller/appcontroller.go]: I think we only need to keep track the first error, as opposed to poErr and savedErr. Basically if *any* error occurs, we will set it once (and not override it). The first error that occurs (if any), will be the one we return as a result of this function.

**Review by jessesuen — CHANGES_REQUESTED** [review]

File: util/health/health.go [hunk 3]
```diff
@@ -59,17 +59,21 @@ func IsWorse(current, new appv1.HealthStatusCode) bool {
 }
 
 func getIngressHealth(obj *unstructured.Unstructured) (*appv1.HealthStatus, error) {
+	health := appv1.HealthStatus{Status: appv1.HealthStatusProgressing}
 	obj, err := kube.ConvertToVersion(obj, "extensions", "v1beta1")
 	if err != nil {
-		return nil, err
+		health.Status = appv1.HealthStatusUnknown
+		health.StatusDetails = err.Error()
+		return &health, err
```
Comment (jessesuen) [inline util/health/health.go]: It looks like we are repeating this logic over and over again. It's better to move the setting of the Status and StatusDetails to the caller of all of these functions if it returns an error.

**Comment by JazminGonzalez-Rivero** [comment]
@jessesuen addressed issues mentioned.

**Review by jessesuen — APPROVED** [review]

**Review by alexmt — APPROVED** [review]

**Pull request merged**